ThingIFSDK/ThingIFSDK/ThingIFAPI.swift

```diff
@@ -146,6 +148,90 @@ public class ThingIFAPI: NSObject, NSCoding {
         }
     }
 
+    /** Endpoints execute onboarding for the thing and merge MQTT channel to the gateway.
+     Thing act as Gateway is already registered and marked as Gateway.
+    
+     - Parameter pendingEndnode: Pending End Node
+     - Parameter endnodePassword: Password of the End Node
+     - Parameter completionHandler: A closure to be executed once on board has finished. The closure takes 2 arguments: an end node, an ThingIFError
+     */
+    public func onboardEndnodeWithGateway(
+        pendingEndnode:PendingEndNode,
+        endnodePassword:String,
+        completionHandler: (EndNode?, ThingIFError?)-> Void
+        ) ->Void
+    {
+        if (self.target == nil) || !(self.target is Gateway) {
+            completionHandler(nil, ThingIFError.TARGET_NOT_AVAILABLE)
+            return
+        }
+        if pendingEndnode.vendorThingID == nil || pendingEndnode.vendorThingID!.isEmpty {
+            completionHandler(nil, ThingIFError.UNSUPPORTED_ERROR)
+            return
+        }
+        if endnodePassword.isEmpty {
+            completionHandler(nil, ThingIFError.UNSUPPORTED_ERROR)
+            return
+        }
+
+        let requestURL = "\(self.baseURL)/thing-if/apps/\(self.appID)/onboardings"
+
+        // generate header
+        let requestHeaderDict:Dictionary<String, String> = [
+            "x-kii-appid": self.appID,
+            "x-kii-appkey": self.appKey,
+            "authorization": "Bearer \(self.owner.accessToken)",
+            "Content-Type": "application/vnd.kii.OnboardingEndNodeWithGatewayThingID+json"
+        ]
+
+        // genrate body
+        let requestBodyDict = NSMutableDictionary(dictionary:
+            [
+                "gatewayThingID": self.target!.typedID.id,
+                "endNodeVendorThingID": pendingEndnode.vendorThingID!,
+                "endNodePassword": endnodePassword,
+                "owner": self.owner.typedID.id
```

Comment:
must use `self.owner.typeID.toString()`, because the value need to be like `user:XXXX`